pi_top_usb_setup/app_fs.py

```
@@ -177,6 +185,65 @@ def set_network(network_data):
+            return
+
+        try:
+            print_folder_recursively(self.FOLDER_TO_COPY)
+        except Exception as e:
+            logger.error(f"Error listing files in {self.FOLDER_TO_COPY}: {e}")
+
+        logger.info(f"Copying files from {self.FOLDER_TO_COPY} to {self.MOUNT_POINT}")
+        for root, dirs, files in walk(self.FOLDER_TO_COPY):
+            for file in files:
+                source_path = path.join(root, file)
+
+                # Get the relative path from the source_dir
+                relative_path = path.relpath(source_path, self.FOLDER_TO_COPY)
+
+                # Destination path based on root
+                destination_path = path.join("/", relative_path)
+
+                # Create target directory if it doesn't exist
+                makedirs(path.dirname(destination_path), exist_ok=True)
+
+                # Copy file
+                copy2(source_path, destination_path)
+                logging.info(f"Copied: {source_path} to {destination_path}")
+
+        if callable(on_progress):
+            on_progress(100.0)
+
+    def run_scripts(self, on_progress: Optional[Callable] = None) -> None:
+        if not Path(self.SCRIPTS_FOLDER).exists():
+            logger.info("No scripts to run; skipping...")
+            return
+
+        try:
+            print_folder_recursively(self.SCRIPTS_FOLDER)
+        except Exception as e:
+            logger.error(f"Error listing files in {self.SCRIPTS_FOLDER}: {e}")
+
+        logger.info(f"Running scripts from {self.SCRIPTS_FOLDER}")
+
+        filenames = sorted(listdir(self.SCRIPTS_FOLDER))
+        for i, file in enumerate(filenames):
+            f = path.join(self.SCRIPTS_FOLDER, file)
+            if path.exists(f):
+                try:
+                    logger.info(f"Executing script: {file} ...")
+                    run_command(f, timeout=60)
+                except Exception as e:
+                    logger.error(f"{e}")
```

i think we should throw this error - no continuing if their setup file is bad

---

"There was an error during setup: E6. Press any button to exit."